### DISCUSSION TITLE
Know when session expired

### DISCUSSION THREAD

**Discussion by tiago-anchorage** [fastlane/fastlane#17965] [General]
I am using Appstore Connect API for almost everything, but unfortunally download_dsyms is using the old method 😛 .

I am trying to send a slack message when we need to renew the session, but I am not finding a way to either: detect that we are waiting for the user input or check that the session expired.

My first (naive) attempt was to catch errors from download_dsyms, but this isn't an "error". An error occurs in the CI after it times out, but I wonder if I can solve this in fastlane alone 🤔

**Comment by rogerluan** [comment]
Hey @tiago-anchorage 👋 

This could be solved by pooling any random `GET` API from time to time, and whenever it fails, send a notification to Slack. What happens exactly when your `download_dsyms` lane fails due to an expired session? 🤔 Sounds like it hangs for awhile waiting for input or something? 2FA code?

**Reply by tiago-anchorage** [reply]
Yes, it hangs waiting for the 2FA code. 

I ended up using a similar tactic that Spaceship does under the hood

```ruby
def check_session() 
  tunesClient = Spaceship::TunesClient.new
  if tunesClient.load_session_from_env
    begin
       return true if tunesClient.fetch_olympus_session
    rescue
        send_invalid_session_message()
    end
  end
end
```

However the session is expiring daily, so eventually I will need to find other solution to sync dSYMs 🙈
